src/struct.cpp

```diff
@@ -118,25 +118,30 @@ void StructManager::Add(const std::string &name, size_t size)
   if (struct_map_.find(name) != struct_map_.end())
     throw std::runtime_error("Type redefinition: type with name \'" + name +
                              "\' already exists");
-  struct_map_[name] = std::make_shared<Struct>(size);
+  struct_map_[name] = std::make_unique<Struct>(size);
 }
 
-std::shared_ptr<Struct> StructManager::Lookup(const std::string &name) const
+Struct *StructManager::Lookup(const std::string &name) const
 {
   auto s = struct_map_.find(name);
-  return s != struct_map_.end() ? s->second : nullptr;
+  return s != struct_map_.end() ? s->second.get() : nullptr;
 }
 
-std::shared_ptr<Struct> StructManager::LookupOrAdd(const std::string &name,
-                                                   size_t size)
+Struct *StructManager::LookupOrAdd(const std::string &name, size_t size)
 {
-  auto s = struct_map_.insert({ name, std::make_shared<Struct>(size) });
-  return s.first->second;
+  auto s = struct_map_.insert({ name, std::make_unique<Struct>(size) });
```

Comment:
Not related to this PR but just noticed: can't `LookupOrAdd()` overwrite existing data? Should we add an error check here?

Reply:
It can't, `std::map::insert` inserts a new element only if the key is not in the map. Otherwise, it just returns an iterator to the existing element.